powershell/VstsTaskSdk/FindFunctions.ps1

```diff
@@ -625,7 +625,7 @@ function Get-FindResult {
 
     if (!(Test-Path -LiteralPath $Path)) {
         Write-Verbose 'Path not found.'
-        return
+        return ,@( )
```

Comment:
what about instead changing the line:
`$findResults = Get-FindResult -Path $findPath -Options $FindOptions`

to:
`$findResults = @( Get-FindResult -Path $findPath -Options $FindOptions )`

I see the caller goes on to print `$findResults.count`. When only one result is returned, `$findResult` will be a string and not an array. I wonder if the string gets coerced to an array when passed to the Filter method, or whether it also errors.

Reply:
> I see the caller goes on to print `$findResults.count`. When only one result is returned, `$findResult` will be a string and not an array. I wonder if the string gets coerced to an array when passed to the Filter method, or whether it also errors.

Actually, `$findResults.count` is not an issue because even if it is `$null`, it returns `$null`. On the opposite, `$findResults` was not coerced to an array when passed to the Filter method and containing `$null`. That was the bug I had. In such case, it throws a `Value cannot be null. Parameter name: source` probably because of a guard in `Filter` method. See the unit test I added to illustrate that: `Find-Match.SupportsPathNotFound.ps1`.

Otherwise I'm fine with your proposal, the bug I had is still fixed and it seems indeed more powershell spirit than my proposal.